### DISCUSSION TITLE
Workflow...

### DISCUSSION THREAD

**Discussion by lovelydinosaur** [encode/httpcore#736] [General]
I'd like to make sure that our workflow on this project and `httpx` is really clearly defined...

* Start with discussions.
* Escalate to an issue *if* there's a consensus on doing so.
* Raise a pull request to resolve the issue.

This'll help keep things nice and clean and ensure that issues & PRs reflect accepted-actionable-work or work-in-progress.

**Comment by karpetrosyan** [comment]
Can we define workflow more precisely? There are some questions for which I do not have clear answers.

- How to recognize when it is time to open the issue from discussion?
- Is it always necessary? What if there is a bug that does not necessitate discussion? (Can we skip discussion -> issue ?)
- Is it necessary to discuss/issue a very minor bug that we can fix with a single line? Isn't it possible to just open a pull request and fix it? (Can we skip discussion -> issue and issue -> pull request and just open the pull request)
- Is it always acceptable to open a pull request when an issue exists, or do we need to discuss pull requests in some cases?

**Reply by lovelydinosaur** [reply]
I think the actionable out of this is that we ought to add issue templates and pull request templates to this project.

We currently have those for `httpx`. Copying the same style to this project seems reasonable.

Would anyone like to open an issue so we can track that, and get them added in?

---

> How to recognise [...]

At a minimum a discussion needs a project maintainer to have escalated it.

> Is it always necessary [...]

For the sake of keeping the issue space / PR space clean I think it's a worthwhile constraint.

> A very minor bug [...]

Typos in the documentation I could see being okay for drive-by PRs? Changes in code behaviour or typing, nah.

> Open a pull request when an issue exists.

Sure.
